deepchecks/vision/checks/distribution/train_test_label_drift.py

```diff
@@ -0,0 +1,447 @@
+    return areas
+
+
+def count_num_bboxes(label):
+    num_bboxes = label.shape[0]
+    return num_bboxes
+
+
+def count_custom_transform_on_label(dataset: VisionDataset, transform: Callable = lambda x: x):
+    counter = Counter()
+    for i in range(len(dataset._data)):
+        list_of_arrays = next(iter(dataset._data))[1]
+        calc_res = [transform(arr) for arr in list_of_arrays]
+        if len(calc_res) != 0 and isinstance(calc_res[0], list):
+            calc_res = [x[0] for x in sum(calc_res, [])]
+        counter.update(calc_res)
+    return counter
+
+
+def histogram_in_batch(dataset: VisionDataset, transform: Callable = lambda x: x):
+    label_min = np.inf
+    label_max = -np.inf
+    for i in range(len(dataset._data)):
+        list_of_arrays = next(iter(dataset._data))[1]
+        calc_res = [transform(arr) for arr in list_of_arrays]
+        if len(calc_res) != 0 and isinstance(calc_res[0], list):
+            calc_res = [x[0] for x in sum(calc_res, [])]
+        label_min = min(calc_res + [label_min])
+        label_max = max(calc_res + [label_max])
+
+    hist, edges = np.histogram([], bins=100, range=(label_min, label_max))
+
+    for i in range(len(dataset._data)):
+        list_of_arrays = next(iter(dataset._data))[1]
+        calc_res = [transform(arr) for arr in list_of_arrays]
+        if len(calc_res) != 0 and isinstance(calc_res[0], list):
+            calc_res = [x[0] for x in sum(calc_res, [])]
+        new_hist, _ = np.histogram(calc_res, bins=100, range=(label_min, label_max))
+        hist = new_hist + hist
+
+    return {k: v for k, v in zip(edges, hist)}
+
+
+PSI_MIN_PERCENTAGE = 0.01
+
+def psi(expected_percents: np.ndarray, actual_percents: np.ndarray):
+    """
+    Calculate the PSI (Population Stability Index).
+
+    See https://www.lexjansen.com/wuss/2017/47_Final_Paper_PDF.pdf
+
+    Args:
+        expected_percents: array of percentages of each value in the expected distribution.
+        actual_percents: array of percentages of each value in the actual distribution.
+
+    Returns:
+        psi: The PSI score
+
+    """
+    psi_value = 0
+    for i in range(len(expected_percents)):
+        # In order for the value not to diverge, we cap our min percentage value
+        e_perc = max(expected_percents[i], PSI_MIN_PERCENTAGE)
+        a_perc = max(actual_percents[i], PSI_MIN_PERCENTAGE)
+        value = (e_perc - a_perc) * np.log(e_perc / a_perc)
+        psi_value += value
+
+    return psi_value
+
+
+def earth_movers_distance_by_histogram(expected_percents: np.ndarray, actual_percents: np.ndarray):
+    dirt = copy(actual_percents)
+    delta = 1 / expected_percents.size
+    emd = 0
+    for i in range(dirt.shape[0]-1):
+        dirt_to_pass = dirt[i] - expected_percents[i]
+        dirt[i+1] += dirt_to_pass
+        emd += abs(dirt_to_pass)*delta
+    return emd
+
+
+def calc_drift_and_plot(train_distribution: dict, test_distribution: dict, plot_title: Hashable,
```

Comment:
A lot of code from here on should be in base right? So we can use it in both subpackages... at least the plotting part, and maybe the computation past the generation of hists

Reply:
yes, will move later